platform-sdk/swirlds-virtualmap/src/main/java/com/swirlds/virtualmap/datasource/VirtualDataSourceBuilder.java

```diff
@@ -37,6 +38,20 @@
  */
 public interface VirtualDataSourceBuilder<K extends VirtualKey, V extends VirtualValue> extends SelfSerializable {
 
+    /**
+     * Creates a new data source builder to use, when a virtual root node is deserialized. Such builder
+     * can't be used to create new data sources, since it lacks virtual map configuration and types,
+     * but it can copy and restore existing data sources.
+     */
+    @SuppressWarnings({"rawtypes", "unchecked"})
+    static <K extends VirtualKey, V extends VirtualValue> VirtualDataSourceBuilder<K, V> newDeserializedBuilder() {
+        final ServiceLoader<VirtualDataSourceBuilder> services = ServiceLoader.load(VirtualDataSourceBuilder.class);
+        for (final VirtualDataSourceBuilder builder : services) {
```

Comment:
We should define some best practices how ServiceLoader should be used. Have it on my todo-list to create a doc. In this code you for example do not check if more than 1 impl is available.